src/components/FeatureViews/ManagePayments/MyPaymentHistory.js

```diff
@@ -0,0 +1,114 @@
+import React, {useEffect, useState} from "react";
+import PaymentTable from "../Accounts/TabComponents/PaymentTable";
+import gql from "graphql-tag";
+import {useSelector} from "react-redux";
+import {useLazyQuery} from "@apollo/react-hooks";
+import moment from "moment";
+import Loading from "../../OmouComponents/Loading";
+import Grid from "@material-ui/core/Grid";
+import Button from "@material-ui/core/Button";
+import {omouBlue} from "../../../theme/muiTheme";
+import CalendarIcon from "@material-ui/icons/CalendarToday";
+import Moment from "react-moment";
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import {DateRange} from "react-date-range";
+import DialogActions from "@material-ui/core/DialogActions";
+import Dialog from "@material-ui/core/Dialog";
+
+export const GET_PARENT_PAYMENTS_FILTERED = gql`
+    query ParentPayments($parentId: ID!, $startDate: String, $endDate: String) {
+        payments(parentId: $parentId, startDate: $startDate, endDate: $endDate) {
+            id
+            createdAt
+            registrationSet {
+              id
+            }
+            total
+            method
+        }
+    }
+`;
+
+export default function MyPaymentHistory() {
+	const AuthUser = useSelector(({auth}) => auth);
+	const [getPayments, {loading, data}] = useLazyQuery(GET_PARENT_PAYMENTS_FILTERED);
+	const [openCalendar, setOpenCalendar] = useState(false);
+	const [state, setState] = useState([
+		{
+			startDate: moment().subtract(1, "month").toDate(),
+			endDate: moment().toDate(),
+			key: 'selection'
+		}
+	]);
+
+	useEffect(() => {
+		getPayments({
+			variables: {
+				parentId: AuthUser.user.id,
+				startDate: moment().subtract(1, "month").toISOString(),
+				endDate: moment().toISOString(),
+			}
+		})
+	}, []);
+
+	const handleDateRangeCalendarChange = (item) => {
+		const newDateRange = item.selection;
+		setState([newDateRange]);
+	}
+
+	const handleSaveDateRange = () => {
+		setOpenCalendar(false);
+		getPayments({
+			variables: {
+				parentId: AuthUser.user.id,
+				startDate: state[0].startDate.toISOString(),
+				endDate: state[0].endDate.toISOString(),
+			},
+		});
+	}
+
+	if (loading || !data) return <Loading/>;
```

Comment:
Why is `!data` checked here? If you're looking for errors, make that explicit (and return a different component)

Reply:
since the data is fetched in a useEffect, data isn't isn't fetched on the first render. it goes past this. lmk if you know a better way of handling this

Reply:
Check the `called` attribute perhaps